tests/components/unifiprotect/test_config_flow.py

```diff
@@ -1022,3 +1034,889 @@ async def test_discovery_can_be_ignored(hass: HomeAssistant) -> None:
+    assert ufp_reconfigure_entry.unique_id == _async_unifi_mac_from_hass(MAC_ADDR)
+    assert ufp_reconfigure_entry.data[CONF_HOST] == "1.1.1.1"
+
+
+async def test_reconfigure_wrong_nvr(
+    hass: HomeAssistant,
+    bootstrap: Bootstrap,
+    nvr: NVR,
+    mock_api_bootstrap: Mock,
+    mock_api_meta_info: Mock,
+) -> None:
+    """Test reconfiguration flow aborts when connected to wrong NVR."""
+    # Use the NVR's actual MAC address
+    nvr_mac = _async_unifi_mac_from_hass(nvr.mac)
+
+    mock_config = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_HOST: "1.1.1.1",
+            CONF_USERNAME: "test-username",
+            CONF_PASSWORD: "test-password",
+            CONF_API_KEY: "test-api-key",
+            "id": "UnifiProtect",
+            CONF_PORT: 443,
+            CONF_VERIFY_SSL: False,
+        },
+        unique_id=nvr_mac,
+    )
+    mock_config.add_to_hass(hass)
+
+    result = await mock_config.start_reconfigure_flow(hass)
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "reconfigure"
+
+    # Create a different NVR (user connected to wrong device)
+    different_nvr = nvr.model_copy()
+    different_nvr.mac = "112233445566"
+    bootstrap.nvr = different_nvr
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_HOST: "2.2.2.2",
+            CONF_PORT: 443,
+            CONF_VERIFY_SSL: False,
+            CONF_USERNAME: "different-username",
+            CONF_PASSWORD: "different-password",
+            CONF_API_KEY: "different-api-key",
+        },
+    )
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "wrong_nvr"
+    # Verify original config wasn't modified
+    assert mock_config.unique_id == nvr_mac
+    assert mock_config.data[CONF_HOST] == "1.1.1.1"
+
+
+async def test_reconfigure_auth_error(
+    hass: HomeAssistant,
+    bootstrap: Bootstrap,
+    nvr: NVR,
+    ufp_reconfigure_entry: MockConfigEntry,
+    mock_api_bootstrap: Mock,
+    mock_api_meta_info: Mock,
+) -> None:
+    """Test reconfiguration flow with authentication error."""
+    ufp_reconfigure_entry.add_to_hass(hass)
+
+    result = await ufp_reconfigure_entry.start_reconfigure_flow(hass)
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "reconfigure"
+
+    # Test with password authentication error
+    mock_api_bootstrap.side_effect = NotAuthorized
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_HOST: "1.1.1.1",
+            CONF_PORT: 443,
+            CONF_VERIFY_SSL: False,
+            CONF_USERNAME: "test-username",
+            CONF_PASSWORD: "wrong-password",
+            CONF_API_KEY: "test-api-key",
+        },
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {CONF_PASSWORD: "invalid_auth"}
```

Comment:
Please finish all config flow tests to reach a terminal state (ABORT or CREATE_ENTRY)